posthog/api/csp.py

```diff
@@ -0,0 +1,134 @@
+import json
+import structlog
+
+from posthog.exceptions import generate_exception_response
+from posthog.utils_cors import cors_response
+from posthog.models.utils import uuid7
+
+logger = structlog.get_logger(__name__)
+
+"""
+| Normalized Key        | report-to format                     | report-uri format                  |
+| --------------------- | ------------------------------------ | ---------------------------------- |
+| `document_url`        | `body.documentURL`                   | `csp-report.document-uri`          |
+| `referrer`            | `body.referrer`                      | `csp-report.referrer`              |
+| `violated_directive`  | *inferred from* `effectiveDirective` | `csp-report.violated-directive`    |
+| `effective_directive` | `body.effectiveDirective`            | `csp-report.effective-directive`   |
+| `original_policy`     | `body.originalPolicy`                | `csp-report.original-policy`       |
+| `disposition`         | `body.disposition`                   | `csp-report.disposition`           |
+| `blocked_url`         | `body.blockedURL`                    | `csp-report.blocked-uri`           |
+| `line_number`         | `body.lineNumber`                    | `csp-report.line-number`           |
+| `column_number`       | `body.columnNumber`                  | *not available*                    |
```

Comment:
csp-report.column-number does seem to exist but only some of the time, see here https://posthog.slack.com/archives/C08S18PDPUN/p1747240708586329 it's in connect-src

Reply:
Nice! I will update it. We're currently checking the ingestion risks and checking the tests so if you see weird pushes, that's it @robbie-c